File: SpringServer/spring/src/main/java/com/springbackend/service/ExerciseService.java
package com.springbackend.service;

import com.springbackend.dto.ExerciseDTO;
import com.springbackend.entity.Exercise;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ExerciseService {
    ExerciseDTO getExercise(String exerciseName);
}

File: CapAndroid/android/app/src/main/java/com/team6/routineapp/service/ExerciseService.kt
package com.team6.routineapp.service

import com.team6.routineapp.dto.ExerciseDTO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ExerciseService {
    @GET("exercises/{exerciseName}")
    fun getExercise(@Path("exerciseName") exerciseName: String): Call<ExerciseDTO>
}